node/agent.go

```diff
@@ -322,8 +331,35 @@ func (a *HostAgent) setupContainer(masterClient master.ClientInterface, svc *ser
 				}
 				cfg.ExposedPorts[dockerclient.Port(p)] = struct{}{}
 				hcfg.PortBindings[dockerclient.Port(p)] = append(hcfg.PortBindings[dockerclient.Port(p)], dockerclient.PortBinding{})
+
+				var assignedPortNumber uint16
+				if a := endpoint.GetAssignment(); a != nil {
+					assignedIP = endpoint.AddressAssignment.IPAddr
+					static = endpoint.AddressAssignment.AssignmentType == commons.STATIC
+					assignedPortNumber = a.Port
+				}
+
+				// set the export data
+				state.Exports = append(state.Exports, zkservice.ExportBinding{
+					Application:        endpoint.Application,
+					Protocol:           endpoint.Protocol,
+					PortNumber:         endpoint.PortNumber,
+					AssignedPortNumber: assignedPortNumber,
+				})
+			} else {
+				state.Imports = append(state.Imports, zkservice.ImportBinding{
+					Application:    endpoint.Application,
+					Purpose:        endpoint.Purpose,
+					PortNumber:     endpoint.PortNumber,
+					PortTemplate:   endpoint.PortTemplate,
+					VirtualAddress: endpoint.VirtualAddress,
+				})
 			}
 		}
+		state.AssignedIP = assignedIP
+		state.Static = static
+	} else {
+		log.Warn("svc.endpoints was nil.")
```

Comment:
I don't think this is worth warning about--in fact this is a case that will never happen